stack/deprecating-interfaces.rst

```diff
@@ -0,0 +1,66 @@
+######################
+Deprecating Interfaces
+######################
+
+Deprecation Procedure
+=====================
+
+Deprecations are changes to interfaces.
+(If they were limited to implementation alone, they wouldn't require deprecation, as no one would notice when the change was made.)
+As a result they often require or result from RFCs.
+But our usual policy applies; if no one would object, a deprecation can be made without an RFC.
+
+Release notes
+-------------
+
+When a feature, class, or function is deprecated, a description of the deprecation is included in the release notes for the next official release (major or minor or bugfix).
+It may be desirable to repeat that description in succeeding release notes until the code is removed.
+
+Code removal
+------------
+
+Removal of the deprecated code occurs, at the earliest, immediately after the next major release following the release with the first deprecation release note; at the latest, immediately before the major release following.
+In other words, if deprecation is first noted in release 17.2.3, the code cannot be removed until after 18.0 is released and must be removed before 19.0 is released.
+In general, no deprecation should be removed before two calendar months have elapsed.
```

Comment:
Can we expand on what this means in practice?

Specifically, is this an assertion that if some developer deprecates an API in release 17.2.3, it is incumbent upon them to remove it before 19.0? Upon their manager? Upon the DM Release Manager?

I guess what I'm saying is that as written this sounds aspirational rather than actionable.

Reply:
Added text about creating a story, blocking the release, and having T/CAM schedule as normal, referencing the appropriate Dev Guide section.